clippy_lints/src/methods/mod.rs

```diff
@@ -1921,43 +1984,18 @@ fn lint_filter_map_map<'a, 'tcx>(
     _filter_args: &'tcx [hir::Expr],
     _map_args: &'tcx [hir::Expr],
 ) {
-    // lint if caller of `.filter().map()` is an Iterator
-    if match_trait_method(cx, expr, &paths::ITERATOR) {
-        let msg = "called `filter_map(p).map(q)` on an `Iterator`. \
-                   This is more succinctly expressed by only calling `.filter_map(..)` instead.";
-        span_lint(cx, FILTER_MAP, expr.span, msg);
-    }
-}
-
-/// lint use of `filter().flat_map()` for `Iterators`
-fn lint_filter_flat_map<'a, 'tcx>(
```

Comment:
@mikerite I added a commit to this PR that removes the linting of `filter().flat_map()` since I couldn't think of a simpler way to achieve the same behavior based on the example you gave. I'd be happy to add it back and include it in the updated docstring if you have any guidance for me here. Thanks!